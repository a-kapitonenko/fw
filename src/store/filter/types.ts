import { Frame } from '../frames/types';

export type Field = {
  name: string;
  value: string;
  checked?: boolean;
};

export type Groups = {
  color: Field[];
  width: Field[];
  noseBridge: Field[];
  shape: Field[];
  material: Field[];
};

export const enum FilterActionTypes {
  FETCH_REQUEST = '@@filter/FETCH_REQUEST',
  FETCH_SUCCESS = '@@filter/FETCH_SUCCESS',
  FETCH_ERROR = '@@filter/FETCH_ERROR',
  CHANGE_CHECKED = '@@filter/CHANGE_CHECKED',
};

export interface IFilterState {
  readonly fetching: boolean;
  readonly groups: Groups;
  readonly errors: string;
  readonly query: Object;
  readonly frames: Frame[];
};
