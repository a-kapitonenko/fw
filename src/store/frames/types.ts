export type Frame = {
  upc: number;
  label: string;
  value: string; 
  img: string;
  compatibility?: boolean;
};

export const enum FramesActionTypes {  
  SUBMIT_START = '@@frames/SUBMIT_START',
  SUBMIT_SUCCESS = '@@frames/SUBMIT_SUCCESS',
  SUBMIT_FAILED = '@@frames/SUBMIT_FAILED',
  FETCH_SIMILAR_FRAMES_START = '@@frames/FETCH_SIMILAR_FRAMES_START',
  FETCH_SIMILAR_FRAMES_SUCCESS = '@@frames/FETCH_SIMILAR_FRAMES_SUCCESS',
  FETCH_SIMILAR_FRAMES_FAILED = '@@frames/FETCH_SIMILAR_FRAMES_FAILED',
  CHECK_FRAME_START = '@@frames/CHECK_FRAME_START',
  CHECK_FRAME_SUCCESS = '@@frames/CHECK_FRAME_SUCCESS',
  CHECK_FRAME_FAILED = '@@frames/CHECK_FRAME_FAILED',
  OPEN = '@@frames/OPEN',
  CLOSE = '@@frames/CLOSE',
  SET_SELECTED_FRAME = '@@frames/SET_SELECTED_FRAME',
  CLEAR_SELECTED_FRAME = '@@frame/CLEAR_SELECTED_FRAME',
  CLEAR_RESULT = '@@frames/CLEAR_RESULT',
};

export interface FramesState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly open: boolean;
  readonly selectedFrame: Frame;
  readonly similarFrames: {
    readonly isFetching: boolean;
    readonly errors: string;
    readonly data: Frame[];
  };
};
