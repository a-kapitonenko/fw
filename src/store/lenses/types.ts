export type Lens = {
  name: string;
  value: number;
};

export const enum LensesActionTypes {
  FETCH_LENSES_START = '@@lenses/FETCH_LENSES_START',
  FETCH_LENSES_SUCCESS = '@@lenses/FETCH_LENSES_SUCCESS',
  FETCH_LENSES_FAILED = '@@lenses/FETCH_LENSES_FAILED',

  SAVE_LENS_START = '@@lenses/SAVE_LENS_START',
  SAVE_LENS_SUCCESS = '@@lenses/SAVE_LENS_SUCCESS',
  SAVE_LENS_FAILED = '@@lenses/SAVE_LENS_FAILED',

  CHECK_LENS_START = '@@lenses/CHECK_LENS_START',
  CHECK_LENS_SUCCESS = '@@lenses/CHECK_LENS_SUCCESS',
  CHECK_LENS_FALIED = '@@lenses/CHECK_LENS_FAILED',
};

export interface ILensesState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly lenses: Lens[];
};
