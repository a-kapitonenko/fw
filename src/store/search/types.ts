import { Frame } from '../frames/types';

export const enum SearchActionTypes {
  SEARCH_START = '@@search/SEARCH_START',
  SEARCH_SUCCESS = '@@search/SEARCH_SUCCESS',
  SEARCH_FAILED = '@@search/SEARCH_FAILED',
  SET_SELECTED_FRAMES = '@@search/SET_SELECTED_FRAMES',
  CLEAR_RESULT = '@@search/CLEAR_RESULT',
}

export interface ISearchState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly frames: Frame[];
  readonly selectedFrames: Frame[];
}
