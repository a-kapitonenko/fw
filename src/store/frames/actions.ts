import { action } from 'typesafe-actions';
import { FramesActionTypes, Frame } from './types';
import { Boss } from '../order/types';

export const open = () => action(FramesActionTypes.OPEN);
export const close = () => action(FramesActionTypes.CLOSE);

export const submitStart = (boss: Boss, frame: Frame) => action(FramesActionTypes.SUBMIT_START, { boss, frame });
export const submitSuccess = () => action(FramesActionTypes.SUBMIT_SUCCESS);
export const submitFailed = (message: string) => action(FramesActionTypes.SUBMIT_FAILED, message);

export const fetchSimilarFramesStart = (boss: Boss) => action(FramesActionTypes.FETCH_SIMILAR_FRAMES_START, boss);
export const fetchSimilarFramesSuccess = (frames: Frame[]) => action(FramesActionTypes.FETCH_SIMILAR_FRAMES_SUCCESS, frames);
export const fetchSimilarFramesFailed = (message: string) => action(FramesActionTypes.FETCH_SIMILAR_FRAMES_FAILED, message);

export const checkFrameStart = (boss: Boss) => action(FramesActionTypes.CHECK_FRAME_START, boss);
export const checkFrameSuccess = () => action(FramesActionTypes.CHECK_FRAME_SUCCESS);
export const checkFrameFailed = (message: string) => action(FramesActionTypes.CHECK_FRAME_FAILED, message);

export const setSelectedFrame = (frame: Frame) => action(FramesActionTypes.SET_SELECTED_FRAME, frame);
export const clearSelectedFrame = () => action(FramesActionTypes.CLEAR_SELECTED_FRAME);

export const clearResult = () => action(FramesActionTypes.CLEAR_RESULT);
