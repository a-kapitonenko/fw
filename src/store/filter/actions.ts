import { action } from 'typesafe-actions';
import { FilterActionTypes, Groups } from './types';
import { Boss } from '../order/types';
import { Frame } from '../frames/types';

export const filteringStart = (boss: Boss, query: object) => action(FilterActionTypes.FILTERING_START, { boss, query });
export const filteringSuccess = (frames: Frame[]) => action(FilterActionTypes.FILTERING_SUCCESS, frames);
export const filteringFailed = (message: string) => action(FilterActionTypes.FILTERING_FAILED, message);

export const fetchGroupsStart = () => action(FilterActionTypes.FETCH_GROUPS_START);
export const fetchGroupsSuccess = (groups: Groups) => action(FilterActionTypes.FETCH_GROUPS_SUCCESS, groups);
export const fetchGroupsFailed = (message: string) => action(FilterActionTypes.FETCH_GROUPS_FAILED, message);

export const changeChecked = (type: string, name: string, value: boolean) => action(FilterActionTypes.CHANGE_CHECKED, { type, name, value });
export const clearChecked = () => action(FilterActionTypes.CLEAR_CHECKED);
export const addQuery = (type: string, value: any) => action(FilterActionTypes.ADD_QUERY, { type, value });
export const deleteQuery = (type: string, value: any) => action(FilterActionTypes.DELETE_QUERY, { type, value });
export const clearQuery = () => action(FilterActionTypes.CLEAR_QUERY);
export const clearResult = () => action(FilterActionTypes.CLEAR_RESULT);
