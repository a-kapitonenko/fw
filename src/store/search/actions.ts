import { Dispatch } from 'redux';
import { action } from 'typesafe-actions';

import { SearchActionTypes } from './types';
import { Boss } from '../order/types';
import { Frame } from '../frames/types';

import { getFramesByUpc } from '../../test/frames';

export const fetchRequest = () => action(SearchActionTypes.FETCH_REQUEST);
export const closeRequest = () => action(SearchActionTypes.CLOSE_REQUEST);
export const setErrors = (errors: string) => action(SearchActionTypes.SET_ERRORS, errors);
export const clearErrors = () => action(SearchActionTypes.CLEAR_ERRORS);
export const setFrames = (frames: Frame[]) => action(SearchActionTypes.SET_FRAMES, frames); 
export const setSelectedFrames = (frames: Frame[]) => action(SearchActionTypes.SET_SELECTED_FRAMES, frames);

export const fetchFrames: any = (boss: Boss, upc: string) => (dispatch: Dispatch) => {
  dispatch(fetchRequest());
  
  return new Promise((resolver) => {
    const response = getFramesByUpc(boss, upc);
    
    setTimeout(() => {
      resolver(response);
    }, 1000);
  })
  .then((response: any) => {
    dispatch(closeRequest());

    if (response.success) {
      dispatch(setFrames(response.result));
    } else {
      dispatch(setErrors(response.errors));
    }
  })
  .catch((errors: any) => dispatch(setErrors(errors)))
};
