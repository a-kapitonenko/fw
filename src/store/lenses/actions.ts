import { Dispatch } from 'redux';
import { action } from 'typesafe-actions';

import { LensesActionTypes, Lens } from './types';
import { IOrderState, Prescription, BossTypes } from '../order/types';
import * as orderActions from '../order/actions';

import { getLenses } from '../../test/lenses';
import { saveLensInformation } from '../../test/order';

// export const fetchRequest = () => action(LensesActionTypes.FETCH_REQUEST);
// export const closeReauest = () => action(LensesActionTypes.CLOSE_REQUEST);
export const setErrors = (errors: string) => action(LensesActionTypes.SET_ERRORS, errors);
export const clearErrors = () => action(LensesActionTypes.CLEAR_ERRORS);
export const setLenses = (lenses: Lens[]) => action(LensesActionTypes.SET_LENSES, lenses);

export const fetchLenses: any = (order: IOrderState) => (dispatch: Dispatch) => {
  dispatch(orderActions.fetchRequest());

  return new Promise((resolver) => {
    const response = getLenses(order);
    
    setTimeout(() => {
      resolver(response);
    }, 1000);
  })
  .then((response: any) => {
    dispatch(orderActions.closeRequest());

    if (response.success) {
      dispatch(setLenses(response.lenses));
    } else { 
      dispatch(setErrors(response.errors));
    }
  })
  .catch((errors: any) => dispatch(setErrors(errors)))
};

export const saveLens: any = (prescription: Prescription, lens: Lens) => (dispatch: Dispatch) => {
  dispatch(orderActions.fetchRequest());

  return new Promise((resolver) => {
    const response = saveLensInformation(prescription, lens)

    setTimeout(() => {
      resolver(response);
    }, 1000);
  })
  .then((response: any) => {
    dispatch(orderActions.closeRequest());

    if (response.success) {
      dispatch(orderActions.setRecommendation(response.recommendation));
      dispatch(orderActions.setBoss(BossTypes.LENS, lens));

    } else { 
      dispatch(setErrors(response.errors));
    }
  })
  .catch((errors: any) => dispatch(setErrors(errors)))
};
