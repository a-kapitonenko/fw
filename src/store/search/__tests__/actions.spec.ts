import * as searchActions from '../actions';
import { Boss } from '../../order/types';
import { Frame } from '../../frames/types';
import * as mockData from '../../../mockData';

describe('search actions', () => {
  it('test searchStart action creator', () => {
    const boss: Boss = mockData.boss;
    const upc: string = '82850523738';

    expect(searchActions.searchStart(boss, upc)).toMatchSnapshot();
  });

  it('test searchSuccess action creator', () => {
    const frames: Frame [] = mockData.frames;

    expect(searchActions.searchSuccess(frames)).toMatchSnapshot();
  });

  it('test searchFailed action creator', () => {
    const message: string = mockData.errorMessage;

    expect(searchActions.searchFailed(message)).toMatchSnapshot();
  });

  it('test setSelectedFrames action creator', () => {
    const frames: Frame [] = mockData.frames;

    expect(searchActions.setSelectedFrames(frames)).toMatchSnapshot();
  });

  it('test clearResult action creator', () => {
    expect(searchActions.clearResult()).toMatchSnapshot();
  });
});
