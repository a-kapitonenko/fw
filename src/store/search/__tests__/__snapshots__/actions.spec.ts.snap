// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search actions test clearResult action creator 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "@@search/CLEAR_RESULT",
}
`;

exports[`search actions test searchFailed action creator 1`] = `
Object {
  "meta": undefined,
  "payload": "Error message",
  "type": "@@search/SEARCH_FAILED",
}
`;

exports[`search actions test searchStart action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "boss": Object {},
    "upc": "82850523738",
  },
  "type": "@@search/SEARCH_START",
}
`;

exports[`search actions test searchSuccess action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Array [
    Object {
      "compatibility": true,
      "img": "1.png",
      "label": "FM17116",
      "upc": 82850553738,
      "value": "82850553738",
    },
    Object {
      "compatibility": false,
      "img": "2.png",
      "label": "SA17346",
      "upc": 82850523999,
      "value": "82850523999",
    },
  ],
  "type": "@@search/SEARCH_SUCCESS",
}
`;

exports[`search actions test setSelectedFrames action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Array [
    Object {
      "compatibility": true,
      "img": "1.png",
      "label": "FM17116",
      "upc": 82850553738,
      "value": "82850553738",
    },
    Object {
      "compatibility": false,
      "img": "2.png",
      "label": "SA17346",
      "upc": 82850523999,
      "value": "82850523999",
    },
  ],
  "type": "@@search/SET_SELECTED_FRAMES",
}
`;
